### PR TITLE
[zh-cn] Update 'working with the tabs api'

### PR DESCRIPTION
**Pull request by jasonren0403**
### Related issues and pull requests

part of #14272

### CODE REVIEW

File: files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md [hunk 1]
```diff
@@ -35,316 +37,305 @@ slug: Mozilla/Add-ons/WebExtensions/Working_with_the_Tabs_API
...
-> - **In Chrome,** the user can select multiple tabs in a window, and the Tabs API sees these as highlighted tabs.
-> - **In Firefox,** the user can't select multiple tabs, so "highlighted" and "active" are synonymous.
-
-### How to example
-
-To see how {{WebExtAPIRef("tabs.query")}} and {{WebExtAPIRef("tabs.Tab")}} are used, let’s walk through how the [tabs-tabs-tabs](https://github.com/mdn/webextensions-examples/tree/main/tabs-tabs-tabs) example adds the list of "switch to tabs" to its toolbar button popup.
+### 示例
 
-![The tabs tabs tabs toolbar menu showing the switch to tap area](switch_to_tab.png)
+为了了解 {{WebExtAPIRef("tabs.query()")}} 和 {{WebExtAPIRef("tabs.Tab")}} 是如何使用的，让我们来看看 [tabs-tabs-tabs](https://github.com/mdn/webextensions-examples/tree/main/tabs-tabs-tabs) 示例是如何在工具栏弹出按钮中添加 “switch to tabs” 列表的。
```
Comment (jasonren0403) [inline files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md:57]:
```suggestion
为了了解 {{WebExtAPIRef("tabs.query()")}} 和 {{WebExtAPIRef("tabs.Tab")}} 是如何使用的，让我们来看看 [tabs-tabs-tabs](https://github.com/mdn/webextensions-examples/tree/main/tabs-tabs-tabs) 示例是如何在工具栏弹出按钮中添加“switch to tabs”列表的。
```

File: files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md [hunk 2]
```diff
@@ -356,231 +347,222 @@ function firstUnpinnedTab(tabs) {
...
+    ```
 
-```js
-function protocolIsApplicable(url) {
-  var anchor = document.createElement("a");
-  anchor.href = url;
-  return APPLICABLE_PROTOCOLS.includes(anchor.protocol);
-}
-```
+    如果示例可以在标签页上执行操作，`initializePageAction()` 会在使 `pageAction` 可见之前将标签页的 `pageAction`（导航栏）图标和标题设置为使用“关闭” 版本：
```
Comment (jasonren0403) [inline files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md:500]:
```suggestion
    如果示例可以在标签页上执行操作，`initializePageAction()` 会在使 `pageAction` 可见之前将标签页的 `pageAction`（导航栏）图标和标题设置为使用“关闭”版本：
```

File: files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md [hunk 3]
```diff
@@ -1,32 +1,34 @@
...
+title: 使用 Tabs API
 slug: Mozilla/Add-ons/WebExtensions/Working_with_the_Tabs_API
+l10n:
+  sourceCommit: 1079b152415f26432481498d2d2b4e8b2f81e3e0
 ---
 
 {{AddonSidebar}}
 
-选项卡允许用户在其浏览器窗口中打开多个网页，然后在这些网页之间切换。使用 Tabs API，你可以使用和操作这些选项卡来创建实用程序，为用户提供使用选项卡或提供扩展功能的新方法。
+选项卡允许用户在其浏览器窗口中打开多个网页，然后在这些网页之间切换。通过使用 Tabs API，你可以使用和操作这些选项卡来创建实用程序，为用户提供使用选项卡或提供扩展功能的新方法。
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md:10]: Could we translate "tab" into "标签页"? As firefox and chromium-based browers have align the translation.

```suggestion
标签页（tab）允许用户在其浏览器窗口中打开多个网页，然后在这些网页之间切换。通过使用 Tabs API，你可以使用和操作这些标签页来创建实用程序，为用户提供使用标签页或提供扩展特性的新方法。
```

File: files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md [hunk 4]
```diff
@@ -1,32 +1,34 @@
...
+- 创建、复制、移动、更新、重新加载和删除标签页。
 - 操纵标签的缩放大小。
-- 操纵选项卡的 CSS。
+- 操纵标签页的 CSS 样式。
 
-最后，我们通过查看 API 提供的其他一些其他功能。
+最后，我们来看看应用程序接口提供的一些其他杂项功能。
 
-> **备注：** 其他地方有一些 Tab API 功能。这些是可用于使用脚本操作选项卡内容的方法{{WebExtAPIRef("tabs.connect")}}，{{WebExtAPIRef("tabs.sendMessage")}}和{{WebExtAPIRef("tabs.executeScript")}}。如果你需要有关这些方法的更多信息，请参阅概念文章[内容脚本](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/Content_scripts)和操作指南[修改网页](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/Modify_a_web_page)。
+> **备注：** 其他地方也介绍了一些 Tabs API 功能。这些方法可用于使用脚本操作标签内容（{{WebExtAPIRef("tabs.connect")}}、{{WebExtAPIRef("tabs.sendMessage")}} 和 {{WebExtAPIRef("tabs.executeScript")}}）。如需了解有关这些方法的更多信息，请参阅概念文章[内容脚本](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/Content_scripts)和操作指南[修改网页](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/Modify_a_web_page)。
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md:22]:
```suggestion
> **备注：** 其他地方也介绍了一些 Tabs API 特性。这些方法可用于使用脚本操作标签页的内容（{{WebExtAPIRef("tabs.connect")}}、{{WebExtAPIRef("tabs.sendMessage")}} 和 {{WebExtAPIRef("tabs.executeScript")}}）。如需了解有关这些方法的更多信息，请参阅概念文章[内容脚本](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/Content_scripts)和操作指南[修改网页](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/Modify_a_web_page)。
```

File: files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md [hunk 5]
```diff
@@ -1,32 +1,34 @@
...
 
-## 权限和选项卡 API
+## 权限和 Tabs API
 
 对于大多数 Tabs API 函数，你不需要任何权限; 但是，有一些例外：
 
-- 需要`"tabs"`权限才能访问 Tab 对象的`Tab.url`，`Tab.title`和`Tab.favIconUrl`属性。在 Firefox 中，你还需要`"tabs"`来按 URL 执行[查询](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/API/tabs/query)。
-- {{WebExtAPIRef("tabs.executeScript")}}或{{WebExtAPIRef("tabs.insertCSS")}}需要[主机权限](/zh-CN/Add-ons/WebExtensions/manifest.json/permissions#Host_permissions)。
+- 需要 `"tabs"` 权限才能访问 Tab 对象的 `Tab.url`、`Tab.title` 和 `Tab.favIconUrl` 属性。在 Firefox 中，你还需要 `"tabs"` 来按 URL 执行[查询](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/API/tabs/query)。
+- {{WebExtAPIRef("tabs.executeScript()")}} 或 {{WebExtAPIRef("tabs.insertCSS()")}} 需要[主机权限](/zh-CN/Add-ons/WebExtensions/manifest.json/permissions#主机权限)。
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md:29]:
```suggestion
- {{WebExtAPIRef("tabs.executeScript()")}} 或 {{WebExtAPIRef("tabs.insertCSS()")}} 需要[主机权限](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions#主机权限)。
```

File: files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md [hunk 6]
```diff
@@ -35,316 +37,305 @@ slug: Mozilla/Add-ons/WebExtensions/Working_with_the_Tabs_API
 ],
 ```
 
-此请求允许你在用户访问的所有网站上使用所有标签 API 功能。还有一种替代方法可以请求权限使用{{WebExtAPIRef("tabs.executeScript")}}或{{WebExtAPIRef("tabs.insertCSS")}}，其中你不需要主机权限，形式为[`"activeTab"`](/zh-CN/Add-ons/WebExtensions/manifest.json/permissions#activeTab_permission)。此权限与`<all_urls>`提供与`"tabs"`相同的权限，但有两个限制：
+此请求允许你在用户访问的所有网站上使用所有 Tabs API 功能。在不需要主机权限的情况下，这里还有一种请求使用 {{WebExtAPIRef("tabs.executeScript()")}} 或 {{WebExtAPIRef("tabs.insertCSS()")}} 的权限的替代方法，即 [`"activeTab"`](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions#活动标签权限) 形式。该权限提供的权限与带有 `<all_urls>` 的 `"tabs"` 相同，但有两个限制：
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md:40]:
```suggestion
此请求允许你在用户访问的所有网站上使用所有 Tabs API 特性。在不需要主机权限的情况下，这里还有一种请求使用 {{WebExtAPIRef("tabs.executeScript()")}} 或 {{WebExtAPIRef("tabs.insertCSS()")}} 的权限的替代方法，即 [`"activeTab"`](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions#活动标签权限) 形式。该权限提供的权限与带有 `<all_urls>` 的 `"tabs"` 相同，但有两个限制：
```

File: files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md [hunk 7]
```diff
@@ -35,316 +37,305 @@ slug: Mozilla/Add-ons/WebExtensions/Working_with_the_Tabs_API
...
     tabsList.appendChild(currentTabs);
   });
 }
 ```
 
-#### Working with the active tab
+#### 使用活动标签页
 
-Another related example feature is the "Alert active tab" info option that dumps all the {{WebExtAPIRef("tabs.Tab")}} object properties for the active tab into an alert:
+另一个相关的示例功能是“警报活动标签页”信息选项，它可将活动标签页的所有 {{WebExtAPIRef("tabs.Tab")}} 对象属性转入警报中：
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md:200]:
```suggestion
另一个相关的示例功能是“警报活动标签页”信息选项，它可将活动标签页的所有 {{WebExtAPIRef("tabs.Tab")}} 对象属性输出到警报中：
```

File: files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md [hunk 8]
```diff
@@ -35,316 +37,305 @@ slug: Mozilla/Add-ons/WebExtensions/Working_with_the_Tabs_API
...
 > - {{WebExtAPIRef("tabs.move")}}
 >
-> Whereas the following functions will act on the active tab (if no tab ID is provided):
+> 而以下函数将作用于活动标签页（如果没有提供标签页 `id`）：
 >
 > - {{WebExtAPIRef("tabs.update")}}
 > - {{WebExtAPIRef("tabs.reload")}}
 
-### How to example
+### 示例
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md:254]: Consider this translation

```suggestion
### 教程示例
```

File: files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md [hunk 9]
```diff
@@ -356,231 +347,222 @@ function firstUnpinnedTab(tabs) {
...
 
-- **default zoom level:** 100%.
-- **zoom mode:** automatic (so the browser manages how zoom levels are set).
-- **scope of zoom changes:** `"per-origin"`, meaning that when you visit a site again, it takes the zoom level set in your last visit.
+- **默认缩放级别**：100%。
+- **缩放模式**：自动（因此浏览器会管理缩放级别的设置）。
+- **缩放更改的范围**：`"per-origin"`，这意味着当你再次访问网站时，网站会采用上次访问时设置的缩放级别。
 
-### How to example
+### 示例
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md:370]: +1

File: files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md [hunk 10]
```diff
@@ -356,231 +347,222 @@ function firstUnpinnedTab(tabs) {
...
-```
+让我们看看缩放功能是如何实现的。
+
+- manifest.json
+  - 缩放功能：缩放功能都不需要权限，因此 [manifest.json](https://github.com/mdn/webextensions-examples/blob/main/tabs-tabs-tabs/manifest.json) 文件中没有需要突出显示的功能。
+- tabs.html
+  - tabs.html：我们已经讨论过 [`tabs.html`](https://github.com/mdn/webextensions-examples/blob/main/tabs-tabs-tabs/tabs.html) 是如何定义该扩展的选项的，在提供缩放选项方面没有做任何新的或独特的工作。
+- tabs.js
+
+  - ：[`tabs.js`](https://github.com/mdn/webextensions-examples/blob/main/tabs-tabs-tabs/tabs.js) 首先定义了缩放代码中使用的几个常量：
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md:384]:
```suggestion
  - : 缩放功能都不需要权限，因此 [manifest.json](https://github.com/mdn/webextensions-examples/blob/main/tabs-tabs-tabs/manifest.json) 文件中没有需要突出显示的功能。
- tabs.html
  - : 我们已经讨论过 [`tabs.html`](https://github.com/mdn/webextensions-examples/blob/main/tabs-tabs-tabs/tabs.html) 是如何定义该扩展的选项的，在提供缩放选项方面没有做任何新的或独特的工作。
- tabs.js

  - : [`tabs.js`](https://github.com/mdn/webextensions-examples/blob/main/tabs-tabs-tabs/tabs.js) 首先定义了缩放代码中使用的几个常量：
```

File: files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md [hunk 11]
```diff
@@ -356,231 +347,222 @@ function firstUnpinnedTab(tabs) {
...
+## 改变标签页的 CSS 样式
 
-Another significant capability offered by the Tabs API is the ability to manipulate the CSS within a tab—add new CSS to a tab ({{WebExtAPIRef("tabs.insertCSS")}}) or remove CSS from a tab ({{WebExtAPIRef("tabs.removeCSS")}}).
+标签 API 提供的另一项重要功能是操作标签内的 CSS 样式：为标签添加新 CSS 样式（{{WebExtAPIRef("tabs.insertCSS()")}}）或从标签移除 CSS 样式（{{WebExtAPIRef("tabs.removeCSS()")}}）。
 
-This can be useful, for example, if you want to highlight certain page elements or change the default layout of the page.
+例如，如果你想突出显示某些页面元素或更改页面的默认布局，这将非常有用。
 
-### How to example
+### 示例
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md:424]: +1

File: files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md [hunk 12]
```diff
@@ -356,231 +347,222 @@ function firstUnpinnedTab(tabs) {
...
 
-#### manifest.json
+- manifest.json
 
-To use the CSS features you need either:
+  - : [`manifest.json`](https://github.com/mdn/webextensions-examples/blob/main/apply-css/manifest.json) 请求使用 CSS 功能所需的权限。你需要
 
-- `"tabs"` permission and [host permission](/zh-CN/Add-ons/WebExtensions/manifest.json/permissions#Host_permissions) or
-- `"activeTab"` permission.
+    - `"tabs"` 权限和[主机权限](/zh-CN/Add-ons/WebExtensions/manifest.json/permissions#主机权限)；或者
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md:436]:
```suggestion
    - `"tabs"` 权限和[主机权限](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions#主机权限)；或者
```

**Review by yin1999 — COMMENTED** [review]
Some suggestions

File: files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md [hunk 13]
```diff
@@ -356,231 +347,222 @@ function firstUnpinnedTab(tabs) {
...
-- Capture the visible tab content with {{WebExtAPIRef("tabs.captureVisibleTab")}}.
-- Detect the primary language of the content in a tab using {{WebExtAPIRef("tabs.detectLanguage")}}. This could be used, for example, to match the language in your extension’s UI with that of the page it’s running in.
+- 使用 {{WebExtAPIRef("tabs.captureVisibleTab")}} 捕捉可见标签内容。
+- 使用 {{WebExtAPIRef("tabs.detectLanguage")}} 检测标签页内容的主要语言。例如，这可用于将扩展用户界面的语言与运行页面的语言相匹配。
 
-## Learn more
+## 进一步学习
 
-If you want to learn more about the Tabs API, check out:
+如果你想了解有关标签 API 的更多信息，请查阅
```
Comment (jasonren0403) [inline files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md:565]:
```suggestion
如果你想了解有关 Tabs API 的更多信息，请查阅
```

File: files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md (hunk 9 above)
Comment (jasonren0403) [reply files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md:370]:
```suggestion
### 教程示例
```

File: files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md (hunk 11 above)
Comment (jasonren0403) [reply files/zh-cn/mozilla/add-ons/webextensions/working_with_the_tabs_api/index.md:424]:
```suggestion
### 教程示例
```

**Review by yin1999 — APPROVED** [review]
Thanks @jasonren0403 :+1:

**Pull request merged**